checkov/arm/checks/SQLServerNoPublicAccess.py

```diff
@@ -19,13 +25,15 @@ def scan_resource_conf(self, conf):
             if conf["resources"]:
                 for resource in conf["resources"]:
                     if "type" in resource:
-                        if resource["type"] == "Microsoft.Sql/servers/firewallRules" or \
-                                resource["type"] == "firewallRules" or \
-                                resource["type"] == "firewallrules":
+                        if (
+                            resource["type"] == "Microsoft.Sql/servers/firewallRules"
+                            or resource["type"].lower() == "firewallrules"
+                        ):
                             if "properties" in resource:
-                                if "startIpAddress" in resource["properties"] and \
-                                        resource["properties"]["startIpAddress"] in ['0.0.0.0', '0.0.0.0/0']:  # nosec
+                                start_ip = resource["properties"].get("startIpAddress")
+                                if start_ip in ["0.0.0.0", "0.0.0.0/0"]:  # nosec
```

Comment:
initially `black` formatted this into: 
```python
if "properties" in resource:
    if "startIpAddress" in resource[
        "properties"
    ] and resource["properties"]["startIpAddress"] in [
        "0.0.0.0",
        "0.0.0.0/0",
    ]:  # nosec
```
`# nosec` don't seem to work here as `bandit` check failed. I've updated it to be in-line. Alternatively it can be set as follows (let me know if you prefer this):
```python
 ] and resource["properties"]["startIpAddress"] in [  # nosec
   ...
```